cmd/logcli/tail.go

```diff
@@ -1,55 +1,21 @@
 package main
 
 import (
-	"time"
-
-	"github.com/grafana/loki/pkg/iter"
-	"github.com/grafana/loki/pkg/logproto"
+	"log"
 )
 
-const tailIteratorIncrement = 10 * time.Second
-
-func tailQuery() iter.EntryIterator {
-	return &tailIterator{
-		from: time.Now().Add(-tailIteratorIncrement),
+func tailQuery() {
+	conn, err := liveTailQueryConn()
+	if err != nil {
+		log.Fatalf("Tailing logs failed: %+v", err)
 	}
-}
-
-type tailIterator struct {
-	from time.Time
-	err  error
-	iter.EntryIterator
-}
-
-func (t *tailIterator) Next() bool {
-	for t.EntryIterator == nil || !t.EntryIterator.Next() {
-		through, now := t.from.Add(tailIteratorIncrement), time.Now()
-		if through.After(now) {
-			time.Sleep(through.Sub(now))
-		}
 
-		resp, err := query(t.from, through, logproto.FORWARD)
+	for {
+		_, message, err := conn.ReadMessage()
 		if err != nil {
-			t.err = err
-			return false
+			log.Println("read:", err)
+			return
 		}
-
-		// We store the through time such that if we don't see any entries, we will
-		// still make forward progress. This is overwritten by any entries we might
-		// see to ensure pagination works.
-		t.from = through
-		t.EntryIterator = iter.NewQueryResponseIterator(resp, logproto.FORWARD)
+		log.Printf("recv: %s", message)
```

Comment:
This would be printing raw json right? Can we have the formatting here?

Reply:
By formatting, do you mean pretty printing with indentation and all or just doing `log.Printf("%s", message)`?